src/_node-compat/headers.ts

```diff
@@ -104,7 +105,12 @@ export const NodeRequestHeaders: {
     }
 
     *entries(): HeadersIterator<[string, string]> {
-      const _headers = this._node.req.headers;
+      const _headers = Object.fromEntries(
```

Comment:
This has performance cost (and memory usage) to iterate in non http2 servers. We should find a way to either subclass or add a flag to skip when not using http2 (benchmarks might not show it)

Reply:
Right, fixed :)